crates/smartmodule-development-kit/src/build.rs

```
@@ -37,9 +37,13 @@ impl BuildOpt {
         if !self.extra_arguments.is_empty() {
             cargo.args(&self.extra_arguments);
         }
-        cargo.status().map_err(Error::from)?;
+        let status = cargo.status().map_err(Error::from)?;
 
-        Ok(())
+        if status.success() {
+            Ok(())
+        } else {
+            Err(anyhow!("Build failed"))
```

We need to include actual errors so they can diagnose not swallow